idl/testing/run_tests.py

```diff
@@ -0,0 +1,1218 @@
+result=          55
+% Compiled module: MDSIDLIMAGE.
+result=          55
+SUCCESS
+
+''')
+
+
+# Issue #2639: mdsvalue, one connect, and kill socket.
+idl_test(f'''
+         
+testid = 'IDL-2639-kill-single-socket'
+PASS = 1
+FAIL = 0
+DATA = '55'
+mdsip_server = '{server}'
+test_status = PASS
+
+result = mdsvalue(DATA)
+print, 'result=', result
+if (result ne fix(DATA)) then test_status = FAIL
+
+mdsconnect, mdsip_server
+mdsdisconnect
+
+result = mdsvalue(DATA)
+print, 'result=', result
+if (result ne fix(DATA)) then test_status = FAIL
+
+if (test_status eq PASS) then begin
+    print, 'SUCCESS'
+end else begin
+    print, 'FAILURE'
+endelse
+
+''',
+'''
+
+% Compiled module: MDSVALUE.
+% Compiled module: MDSCHECKARG.
+% Compiled module: MDSISCLIENT.
+% Compiled module: MDS_KEYWORD_SET.
+% Compiled module: MDSIDLIMAGE.
+result=          55
+% Compiled module: MDSCONNECT.
+% Compiled module: MDSDISCONNECT.
+result=          55
+SUCCESS
+
+''')
+
+
+# Issue #2639: mdsvalue, one connect, and kill 0 socket.
+# Note different behavior from IDL-2639-kill-single-socket test.
+# idl_test(f'''
+         
+# testid = 'IDL-2639-kill-single-zero'
+# PASS = 1
+# FAIL = 0
+# DATA = '55'
+# mdsip_server = '{server}'
+# test_status = PASS
+
+# result = mdsvalue(DATA)
+# print, 'result=', result
+# if (result ne fix(DATA)) then test_status = FAIL
+
+# mdsconnect, mdsip_server
+# mdsdisconnect, socket=0
+
+# result = mdsvalue(DATA)
+# print, 'result=', result
+# if (result ne fix(DATA)) then test_status = FAIL
+
+# if (test_status eq PASS) then begin
+#     print, 'SUCCESS'
+# end else begin
+#     print, 'FAILURE'
+# endelse
+
+# ''',
+# '''
+
+# % Compiled module: MDSVALUE.
+# % Compiled module: MDSCHECKARG.
+# % Compiled module: MDSISCLIENT.
+# % Compiled module: MDS_KEYWORD_SET.
+# % Compiled module: MDSIDLIMAGE.
+# result=          55
+# % Compiled module: MDSCONNECT.
+# % Compiled module: MDSDISCONNECT.
+# result=          55
+# SUCCESS
+
+# ''')
+
+
+# Issue #2640: disconnect returns correct status.
+# First disconnect should succeed and thus return True (1).
+# But disconnecting an already disconnected socket should return False (0).
+# idl_test(f'''
+         
+# testid = 'IDL-2640-status'
+# PASS = 1
+# FAIL = 0
+# BOGUS = -77
+# mdsip_server = '{server}'
+# test_status = PASS
+
+# mdsconnect, mdsip_server
+# print, '!MDS_SOCKET=', !MDS_SOCKET
+
+# statusvar = BOGUS
+# mdsdisconnect, socket=0, status=statusvar
+# print, '!MDS_SOCKET, statusvar=', !MDS_SOCKET, statusvar
+# if (statusvar ne 1) then test_status = FAIL
+
+# statusvar = BOGUS
+# mdsdisconnect, socket=0, status=statusvar
+# print, '!MDS_SOCKET, statusvar=', !MDS_SOCKET, statusvar
+# if (statusvar ne 0) then test_status = FAIL
+
+# if (test_status eq PASS) then begin
+#     print, 'SUCCESS'
+# end else begin
+#     print, 'FAILURE'
+# endelse
+
+# ''',
+# '''
+
+# % Compiled module: MDSCONNECT.
+# % Compiled module: MDS_KEYWORD_SET.
+# % Compiled module: MDSDISCONNECT.
```

Comment:
Why comment out this test?

Reply:
Hi @smithsp -- Excellent questions.

During the course of fixing Issue 2625, my manual testing uncovered additional bugs in the IDL API.   And those bugs have been submitted to GitHub (see Issues 2638, 2639 and 2640).   

When I converted my manual tests into written test cases, I wrote test cases for those additional bugs.

However, those additional bugs have not yet been fixed.   They are not part of Issue 2625, so are in the backlog.

If I left those three tests uncommented, then **every** build of MDSplus would fail from now until whenever those additional IDL bugs are fixed.

I instead added a note to the three additional issues indicating that the test cases have already been written and should be uncommented when the bugs are eventually fixed.

Mention: @joshStillerman @WhoBrokeTheBuild

Reply:
@mwinkel-dev That is a reasonable explanation.  Thank you for documenting the bugs you find.  
(For easier linking, if you put `#2638` in the comment it gets rendered as #2638 .)